ports/mimxrt/hal/pwm_backport.c

```diff
@@ -0,0 +1,198 @@
+/*
+ * This file is part of the MicroPython project, http://micropython.org/
+ *
+ * Copyright (c) 2015, Freescale Semiconductor, Inc.
+ * Copyright 2016-2017 NXP *
+ * SPDX-License-Identifier: BSD-3-Clause
+ *
+ * The MIT License (MIT)
```

Comment:
I'm not sure about making a file have 2 licenses... I think if you copied and adapted the code from the NXP library then it should remain just BSD-3-Clause, and you can add your copyright to that if you like (ie don't add the MIT text).

Reply:
OK. Yes, I took that code from NXP, because the NXP lib did not support duty_u16, and I did not want to loose resolution by scaling up from percent to u16.